lisa/sut_orchestrator/azure/common.py

```diff
@@ -225,6 +354,63 @@ class AzureVmMarketplaceSchema(AzureImageSchema):
     def __hash__(self) -> int:
         return hash(f"{self.publisher}/{self.offer}/{self.sku}/{self.version}")
 
+    def _get_info(self, platform: "AzurePlatform") -> Optional[Dict[str, Any]]:
+        for location in platform._find_marketplace_image_location():
+            image_info = platform._get_image_info(location, self)
+            if image_info:
+                return _get_image_tags(image_info)
+        return None
+
+    def _parse_info(self, raw_features: Dict[str, Any]) -> None:
+        # CPU Architecture
+        arch = raw_features.get("architecture")
+        if arch == "arm64":
+            self.architecture = ArchitectureType.Arm64
+        elif arch == "x64":
+            self.architecture = ArchitectureType.x64
+
+        # Disk Controller Type
+        disk_controller_type = raw_features.get("DiskControllerTypes")
+        if (
+            isinstance(disk_controller_type, str)
+            and disk_controller_type.lower() == "scsi"
+        ):
+            self.disk_controller_type = schema.DiskControllerType.SCSI
+        elif (
+            isinstance(disk_controller_type, str)
+            and disk_controller_type.lower() == "nvme"
+        ):
+            self.disk_controller_type = schema.DiskControllerType.NVME
+
+        # Hyper-V Generation
+        try:
+            gen = raw_features.get("hyper_v_generation")
+            if gen:
+                self.hyperv_generation = int(gen.strip("V"))
+        except (TypeError, ValueError, AttributeError):
+            pass
```

Comment:
Does it need a debug level log?

Reply:
Should I pass in logger from platform? (I'm not super familiar with how loggers are handled)

Reply:
Yes, you can use the logger from platform. The different logs decide the prefix of log lines, and may be put into different log files.